add documentation for the docker container generations scripts (#1129)
pyne/pyne: docker/readme.md

Docker Script Documentation
===========================

The files in this folder are intended to be mainly used by PyNE maintainers,

> they provide an "easy" mechanism to build and push the different containers

Reviewer: ```suggestion
they provide an "easy" mechanism to build and push the various containers
```

> required for the multiple configurations test by the PyNE CI.

Reviewer: ```suggestion
required for the multiple configurations tested by the PyNE CI.
```

This folder contains:

> - A custom Dockerfile: allowing to build docker containers with a custom set of

Reviewer 2: ```suggestion
- A custom Dockerfile: "ubuntu_18.04-dev.dockerfile", allowing to build docker containers with a custom set of
```

> - A custom Dockerfile: "ubuntu_18.04-dev.dockerfile", allowing to build docker containers with a custom set of

Reviewer: ```suggestion
- A custom Dockerfile: "ubuntu_18.04-dev.dockerfile", allowing the user 
to build docker containers with a custom set of
```

>   dependencies required by the CI, the configuration of the Docker containers

Reviewer: ```suggestion
  dependencies required by the CI. The configuration of the Docker containers
```

>   can be specified using the "--build-args" flags. The different flags are:

Reviewer: ```suggestion
  can be specified using the `--build_***` flags. The available flags are:
```

Author: the correct usage in docker is actually `--build_args build_moab=YES`.

maybe `--build_args build_***=YES` is the good way to write it ?
or `--build_args *********` ?

Reviewer: ah -- gotcha... that was not clear. I get it now.

If that's the case, then this place should, indeed, say [`--build-arg <flag>`](https://github.com/pyne/pyne/blob/3374b304124dd345b91c99ed3b8aa8b32fc7d85e/docker/make_pyne_docker_image.py#L66) .

>     - "build_moab=YES" add MOAB in the docker container

Reviewer: ```suggestion
    - `build_moab=YES` add MOAB in the docker container
```

>     - "build_pymoab=YES" install MOAB with pymoab

Reviewer: ```suggestion
    - `build_pymoab=YES` install MOAB with pymoab
```

>     - "build_dagmc=YES" add DAGMC 

Reviewer: ```suggestion
    - `build_dagmc=YES` add DAGMC 
```

>     - "build_pyne=NO" only add the PyNE dependencies to the container

Reviewer: ```suggestion
    - `build_pyne=NO` do not build PyNE, only add the PyNE dependencies to the container
```

>     - "py_version=X.Y" specify the python version to install (option are 2.7 or

Reviewer: ```suggestion
    - `py_version=X.Y` specify the python version to install (options are 2.7 or
```

      3.6)

> - A python script is also present to simplify the usage of the docker file, it

Reviewer 2: ```suggestion
- A python script, "make_pyne_docker_image.py", is also present to simplify the usage of the docker file, it
```

> - A python script, "make_pyne_docker_image.py", is also present to simplify the usage of the docker file, it

Reviewer: ```suggestion
- A python script, `make_pyne_docker_image.py`, is also present to simplify the usage of the docker file. It
```

>   will allows maintainers to build, name and push the docker container into

Reviewer: ```suggestion
 allows maintainers to build, name, and push the docker container into
```

>   DockerHub. The different option are:

Reviewer: ```suggestion
  DockerHub. The available options are:
```

>     - "--py_version=X" with "X" = 2 or 3, specify the python version

Reviewer: ```suggestion
    - `--py_version=X` with "X" = major python version (e.g. 2 or 3)
```

>     - "--moab" install MOAB in the container

Reviewer: ```suggestion
    - `--moab` install MOAB in the container
```

>     - "--dagmc" install DAGMC in the container

Reviewer: ```suggestion
    - `--dagmc` install DAGMC in the container
```

>     - "--pymoab" install pyMOAB in the container (along with MOAB)

Reviewer: ```suggestion
    - `--pymoab` install pyMOAB in the container (along with MOAB)
```

>     - "--all/-a/-all" install all the optional dependencies (MOAB/pyMOAB/DAGMC)

Reviewer: ```suggestion
    - `--all/-a/-all` install all the optional dependencies (MOAB/pyMOAB/DAGMC)
```

>     - "--deps" don't install PyNE only install the selected optional PyNE dependencies (along with the required one)

Reviewer: ```suggestion
    - `--deps` don't install PyNE, only install the dependencies (both required and selected optional dependencies)
```

>     - "--push" push the docker container to DockerHub after building it

Reviewer: ```suggestion
    - `--push` push the docker container to DockerHub after building it
```

      (requires right to push on the PyNE DockerHub account)

>     - "-name=" manually set the docker container name (not recommended with used

Reviewer 2: ```suggestion
    - "--name=" manually set the docker container name (NOT recommended with used
```

>     - "--name=" manually set the docker container name (NOT recommended with used

Reviewer: ```suggestion
    - `--name=` manually set the docker container name (NOT recommended to be used
```

      with the push option)

> The Python script will check consistency in the required dependency to ensure

Reviewer: ```suggestion
The Python script will check for consistency among the required dependencies to ensure
```

> working build: pyMOAB requires MOAB to be install (MOAB will be install if

Reviewer: ```suggestion
working build. For example, pyMOAB requires MOAB to be install (MOAB will be installed if
```

> pyMOAB flag is provided, even if MOAB flag is not), DAGMC requires pyMOAB (and

Reviewer: ```suggestion
the pyMOAB flag is provided, even if the MOAB flag is not), DAGMC requires pyMOAB (and
```

MOAB).

> The Python script will also also form the container name accordingly to some

Reviewer 2: ```suggestion
The Python script will also name container according to some
```

> The Python script will also name container according to some

Reviewer: ```suggestion
The Python script will also name the container according to the
```

> convention allowing the automatic CI to use the pushed docker image, One

Reviewer: ```suggestion
convention, allowing the automatic CI to use the pushed docker image. We
```

> **STRONGLY** advices against manually set the container name when updating the CI

Reviewer: ```suggestion
**STRONGLY** advise against manually setting the container name when updating the CI
```

docker containers in the PyNE DockerHuB. 

> One recommends to maintainers to use the python script to generate and push the docker

Reviewer 2: ```suggestion
We recommend maintainers use the python script to generate and push the docker
```

> container in order to form container name correctly, and include the proper

Reviewer: ```suggestion
container in order to form container name correctly and to include the proper
```

Docker Script Documentation
===========================

The files in this folder are intended to be mainly used by PyNE maintainers,
they provide an "easy" mechanism to build and push the various containers
required for the multiple configurations tested by the PyNE CI.

This folder contains:

- A custom Dockerfile: "ubuntu_18.04-dev.dockerfile", allowing the user 
to build docker containers with a custom set of
  dependencies required by the CI. The configuration of the Docker containers
  can be specified using the `--build-args <flag>` argument. The different flags are:
    

>     - `build_moab=YES` add MOAB in the docker container

Reviewer 3: Are these the default values?  Presumably you can say either `YES` or `NO` for each option?

    - `build_pymoab=YES` install MOAB with pymoab
    - `build_dagmc=YES` add DAGMC 
    - `build_pyne=NO` do not build PyNE, only add the PyNE dependencies to the container
    - `py_version=X.Y` specify the python version to install (options are 2.7 or
      3.6)
Note, if using multiple build arguments `--build-args` has to be repeted, i.e. :

> to request DAGM, dependenciesonly with Python 2.7, the docker build arguments will be: 

Reviewer 2: ```suggestion
to request DAGMC dependencies only with Python 2.7, the docker build arguments will be: 
```

> `--build-args build_dagmc=YES --build-args build_pyne=NO --build-args py_version=2.7` 

Author: Does it look better ?
(I also added an example...)

(thx for the review and the English corrections!)

> `--build-args build_dagmc=YES --build-args build_pyne=NO --build-args=py_version=2.7` 

Author: Does it look better @katyhuff ? 
(I also added an example...)

(thx for the review and the English corrections!)
